pkg/controller/autoimport/apiserverurl.go

```diff
@@ -0,0 +1,103 @@
+package autoimport
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/stolostron/managedcluster-import-controller/pkg/constants"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func updateClusterURL(ctx context.Context, client client.Client,
+	managedCluster *clusterv1.ManagedCluster, autoImportSecret *corev1.Secret) error {
+	apiServerURL, err := getAPIServerURL(autoImportSecret)
+	if err != nil {
+		return err
+	}
+
+	if apiServerURL == "" {
+		return nil
+	}
+
+	apiServerURL = strings.TrimSuffix(apiServerURL, "/")
+
+	clusterCopy := managedCluster.DeepCopy()
+	for _, config := range clusterCopy.Spec.ManagedClusterClientConfigs {
+		if config.URL == apiServerURL {
```

Comment:
should it also trimSuffix "/" for the `config.URL` before comparing? since the `config.URL` might be configured by users?

Reply:
done